x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/FormatterState.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+
+import java.util.List;
+
+public interface FormatterState extends NamedWriteable {
```

Comment:
Why this new interface? You can add directly the formatterstate to `Cursors` list of writeables.